substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/heap/Target_java_lang_ref_PhantomReference.java

```diff
@@ -36,4 +38,8 @@ public final class Target_java_lang_ref_PhantomReference<T> {
     boolean refersTo0(Object obj) {
         return ReferenceInternals.refersTo(SubstrateUtil.cast(this, PhantomReference.class), obj);
     }
+
+    @AnnotateOriginal
+    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
```

Comment:
I understand why you need it for `Reference`, but why do you need to add the uninterruptible annotation to this class?

Reply:
Good question. I don't know the exact details on the limitations around `@Uninterruptible`, but unless you do this native image fails, e.g.:

```
com.oracle.svm.core.util.VMError$HostedError: Found 1 violations of @Uninterruptible usage:
- method java.lang.ref.Reference.get():Object is annotated but java.lang.ref.PhantomReference.get():Object is not
```

I suspect that if classes override a method that is `@Uninterruptible`, e.g. `PhantomReference`, then they also need to mark it `@Uninterruptible`.